src/commands/createFunctionApp/containerImage/detectDockerfile.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { AzExtFsExtra, DialogResponses } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { dockerfileGlobPattern, hostFileName } from '../../../constants';
+import { type ICreateFunctionAppContext } from '../../../tree/SubscriptionTreeItem';
+import { findFiles } from '../../../utils/workspace';
+import { isFunctionProject } from '../../createNewProject/verifyIsProject';
+import path = require('path');
+
+export async function detectDockerfile(context: ICreateFunctionAppContext): Promise<void> {
+    if (vscode.workspace.workspaceFolders) {
+        context.workspaceFolder = vscode.workspace.workspaceFolders[0];
+        let hostPath: string = context.workspaceFolder.uri.fsPath;
+
+        if (await isFunctionProject(context.workspaceFolder.uri.fsPath)) {
+            context.rootPath = context.workspaceFolder.uri.fsPath;
+        } else {
+            context.rootPath = context.workspaceFolder.uri.fsPath;
+            const files = (await findFiles(context.workspaceFolder, `*/${hostFileName}`));
+            hostPath = path.dirname(files[0].fsPath);
+        }
+
+        // check if dockerfile exists in the same folder as the host.json
+        context.dockerfilePath = (await findFiles(hostPath, dockerfileGlobPattern))[0].fsPath;
+
+        // prompt user to proceed with containerized function app creation
+        const placeHolder: string = 'Dockerfile detected. Would you like to continue creating a containerized function app?';
+        const result: vscode.MessageItem = await context.ui.showWarningMessage(placeHolder, { modal: true }, DialogResponses.yes, DialogResponses.no);
```

Comment:
I thought about changing this since I also wasn't sure if the portal team wan't to brand them as "containerized function apps". In the portal when creating a containerized function app you choose to deploy with an image: 
![image](https://github.com/microsoft/vscode-azurefunctions/assets/59709511/14f60110-6a31-41d2-b61e-7361f5c9c6cf)
We could do a similar message like : 
"Dockerfile detected. Would you like to continue creating a function app that deploys an image? Click yes to continue, click no to create a function app that deploys code."

Reply:
Maybe we could do something like.

`Dockerfile with Azure Functions image detected. What would you like to deploy?`
`Code` `Container image`